exporter/exporter.go

```diff
@@ -0,0 +1,160 @@
+	// These calls are important in order to configure the unit of the metrics,
+	// but in some cases all the exported metrics are builtin (unit is configured)
+	// or the unit is not important.
+	SkipCMD bool
+
+	// Timeout for all API calls. If not set, defaults to 5 seconds.
+	Timeout time.Duration
+
+	// ReportingInterval sets the interval between reporting metrics.
+	// If it is set to zero then default value is used.
+	ReportingInterval time.Duration
+
+	// NumberOfWorkers sets the number of go rountines that send requests
+	// to Stackdriver Monitoring. This is only used for Proto metrics export
+	// for now. The minimum number of workers is 1.
+	NumberOfWorkers int
+
+	// MetricDescriptorTypeFormatter is the custom formtter for the MetricDescriptor.Type.
+	// By default, the format string is "custom.googleapis.com/opentelemetry/[metric name]".
+	MetricDescriptorTypeFormatter func(*apimetric.Descriptor) string
+}
+
+
+// Exporter is a trace and metric exporter
+type Exporter struct {
+	traceExporter *cloudtrace.Exporter
+}
+
+
+// NewExporter creates a new Exporter that implements both trace.Exporter
+// and metric.Exporter
+func NewExporter(o Options) (*Exporter, error) {
+
+	te, err := cloudtrace.NewExporter(cloudtrace.WithProjectID(o.ProjectID), cloudtrace.WithContext(o.Context),
+		cloudtrace.WithTraceClientOptions(o.TraceClientOptions), cloudtrace.WithTimeout(o.Timeout), 
+		cloudtrace.WithOnError(o.OnError),
+	)
+	if err != nil {
+		return nil, err
+	} 
+	
+	return &Exporter{
+		traceExporter: te,
+	}, nil
+}
+
+// readMonitoredResourcesFromMetricsExporter obtains Monitored Resources labels from metrics exporter
+func readMonitoredResourcesFromMetricsExporter() (resType string, labels map[string]string) {
+	//TODO: read the resources from metric. Now the autodetect is for mockup
+	return monitoredresource.Autodetect().MonitoredResource()
+}
+
+// injectLabelsIntoSpan injects Monitored Resources labels into a span 
+func injectLabelsIntoSpan(sd *texport.SpanData, labels map[string]string) *texport.SpanData {
+	for k, v := range labels {
+		sd.Attributes = append(sd.Attributes, kv.Key(k).String(v))
+	}
+	return sd
+}
+
+// injectLabelsIntoSpan injects Monitored Resources labels into spans
+func injectLabelsIntoSpans(sds []*texport.SpanData, labels map[string]string) []*texport.SpanData {
+	for i, v := range sds {
+		sds[i] = injectLabelsIntoSpan(v, labels)
+	}
+	return sds
+}
+
+// ExportSpan exports a SpanData to Stackdriver Trace.
+func (e *Exporter) ExportSpan(ctx context.Context, sd *texport.SpanData) {
+	_, monitoredResMap := readMonitoredResourcesFromMetricsExporter()
+	sd = injectLabelsIntoSpan(sd, monitoredResMap)
+	e.traceExporter.ExportSpan(ctx, sd)
+}
+
+// ExportSpans exports a slice of SpanData to Stackdriver Trace in batch
+func (e *Exporter) ExportSpans(ctx context.Context, sds []*texport.SpanData) {
+	_, monitoredResMap := readMonitoredResourcesFromMetricsExporter()
+	sds = injectLabelsIntoSpans(sds, monitoredResMap)
```

Comment:
Seems like this is the main purpose of the wrapper - injecting auto-detected resource labels into spans.
Is that correct?

Reply:
Yes, it is. We want a combined exporter to upload spans with monitored resource labels. Besides, we may modify metric exporter so that it can access and preserve the monitored resource labels and export them to the combined exporter.